src/profile-logic/marker-data.js

```diff
@@ -239,150 +238,283 @@ export function deriveMarkersFromRawMarkerTable(
+              data.interval
+            }' in marker index ${i}. This should not normally happen.`
+          );
+          matchedMarkers.push({
+            start: time,
             dur: 0,
+            name: stringTable.getString(name),
             title: null,
             data,
-            incomplete: true,
-          };
-        }
-        if (marker.start !== undefined) {
-          marker.dur = time - marker.start;
+          });
         }
-        matchedMarkers.push(marker);
+        break;
       }
-    } else if (data.type === 'CompositorScreenshot') {
-      // Screenshot markers are already ordered. We compute their duration using the
-      // following marker of the same type.
-      if (previousScreenshotMarker !== null) {
-        previousScreenshotMarker.dur =
-          rawMarkers.time[i] - previousScreenshotMarker.start;
-        matchedMarkers.push(previousScreenshotMarker);
+
+      case 'Network': {
+        // Network markers are similar to tracing markers in that they also
+        // normally exist in pairs of start/stop markers. But unlike tracing
+        // markers they have a duration and "startTime/endTime" properties like
+        // more generic markers. Lastly they're always adjacent.
+
+        // The timestamps on the start and end markers describe two
+        // non-overlapping parts of the same load. The start marker has a
+        // duration from channel-creation until Start (i.e. AsyncOpen()). The
+        // end marker has a duration from AsyncOpen time until OnStopRequest.
+        // In the merged marker, we want to represent the entire duration, from
+        // channel-creation until OnStopRequest
+
+        if (data.status === 'STATUS_START') {
+          openNetworkMarkers.set(name, i);
+        } else {
+          // End status can be any status other than 'STATUS_START'. They are
+          // either 'STATUS_STOP' or 'STATUS_REDIRECT'.
+          const endData = data;
+
+          const startIndex = openNetworkMarkers.get(name);
+          if (startIndex !== undefined) {
+            // A start marker matches this end marker.
+            openNetworkMarkers.delete(name);
+
+            // We know this startIndex points to a Network marker.
+            const startData: NetworkPayload = (rawMarkers.data[
+              startIndex
+            ]: any);
+
+            matchedMarkers.push({
+              start: startData.startTime,
+              dur: endData.endTime - startData.startTime,
+              name: stringTable.getString(name),
+              title: null,
+              data: {
+                ...endData,
+                startTime: startData.startTime,
+              },
+            });
+          } else {
+            // There's no start marker matching this end marker. This means an
+            // abstract marker exists before the start of the profile.
+            const start = Math.min(firstSampleTime, endData.startTime);
```

Comment:
Observation: This line can make the marker extend beyond how we display the `EmptyThreadIndicator`. The network markers are at least in their own track.

Reply:
I'm not sure of what the implications of this is. If this is an issue, a profile with the problem in a separate issue would be a good idea.